LibreNMS/Util/Oid.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/*
+ * Snmp.php
+ *
+ * -Description-
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @package    LibreNMS
+ * @link       http://librenms.org
+ * @copyright  2022 Tony Murray
+ * @author     Tony Murray <murraytony@gmail.com>
+ */
+
+namespace LibreNMS\Util;
+
+use Cache;
+use LibreNMS\Exceptions\InvalidOidException;
+
+class Oid
+{
+    public static function isNumeric(string $oid): bool
+    {
+        return (bool) preg_match('/^[.\d]+$/', $oid);
+    }
+
+    public static function hasNumeric(array $oids): bool
+    {
+        foreach ($oids as $oid) {
+            if (self::isNumeric($oid)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Converts an oid to numeric and caches the result
+     *
+     * @throws \LibreNMS\Exceptions\InvalidOidException
+     */
+    public static function toNumeric(string $oid, string $mib = 'ALL', int $cache = 1800): string
+    {
+        if (Oid::isNumeric($oid)) {
+            return $oid;
+        }
+        $key = 'Oid:toNumeric:' . $mib . '/' . $oid;
+
+        $numeric_oid = Cache::remember($key, $cache, function () use ($oid, $mib) {
```

Comment:
Any reason to not cache this way longer?
What's the chance a MIB changes OID number? we could flush it after every update also

Reply:
The idea was to ensure that the value is cached all long one single discovery.
And the initial goal was to allow it even in production (so a YAML file could lack num_oid and still be valid).

But currently num_oid is mandatory, so this code is only used for YAML file development.